pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,262 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, premake5
+, writeShellScriptBin
+, runCommandLocal
+, symlinkJoin
+, imagemagick
+, bzip2
+, curl
+, flac
+, fmt_9
+, freetype
+, libevent
+, libgit2
+, libGL
+, libGLU
+, libjpeg
+, libpng
+, libvorbis
+, libX11
+, libxkbcommon
+, libXxf86vm
+, lua5_3
+, nlohmann_json
+, noto-fonts-cjk-sans
+, openal
+, SDL2
+, sqlite
+, wayland
+, egl-wayland
+,
+}:
+let
+  edopro-version = "40.1.4";
+
+  # the assets repo has a font instead
+  font = "${noto-fonts-cjk-sans}/share/fonts/opentype/noto-cjk/NotoSansCJK-VF.otf.ttc";
+
+  covers_url = "";
+  fields_url = "";
+  pics_url = "http://images.ygoprodeck.com/images/cards/{}.jpg";
+
+  assets = {
+    pname = "assets";
+    version = edopro-version;
+
+    src = fetchTarball {
+      url = "https://github.com/ProjectIgnis/edopro-assets/releases/download/${edopro-version}/ProjectIgnis-EDOPro-${edopro-version}-linux.tar.gz";
+      sha256 = "sha256-vZhkWJ1ZoNEwdc5kM1S0hyXnWmupiTOanCi9DCuqw/k=";
+    };
+  };
+
+  irrlicht-edopro = stdenv.mkDerivation {
+    pname = "irrlicht-edopro";
+    version = "1.9-custom";
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = "irrlicht1-8-4";
+      # Found by getting the newest version before EDOPro's tagged release comes out.
+      rev = "7edde28d4f8c0c3589934c398a3a441286bb7c22";
+      hash = "sha256-Q2tNiYE/enZPqA5YhUe+Tkvmqtmmz2E0OqTRUDnt+UA=";
+    };
+
+    buildInputs = [
+      libGLU
+      libX11
+      libxkbcommon
+      libXxf86vm
+      wayland
+    ];
+
+    enableParallelBuilding = true;
+    buildFlags = "NDEBUG=1";
+    makeFlags = "-C source/Irrlicht";
+
+    installPhase = ''
+      mkdir -p $out/{bin,include}
+      cp lib/Linux/libIrrlicht.a $out/bin
+      cp -r include/* $out/include
+    '';
+  };
+
+  ocgcore = stdenv.mkDerivation rec {
+    pname = "ygopro-core";
+    # Tied to the tagged version of EDOPro's git module.
+    version = "4a872ad7686e0d00ee4c1c051e90fc07c40613d4";
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = pname;
+      rev = version;
+      hash = "sha256-s3DOHrZilRUy6fbXObNiLRinxpdHY99vDOmS/ZfOI78=";
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      lua5_3
+    ];
+
+    preBuild = ''
+      premake5 gmake2
+    '';
+
+    enableParallelBuilding = true;
+    buildFlags = "verbose=true config=release ocgcore";
+    makeFlags = "-C build";
+
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/release/libocgcore.a $out/bin
+    '';
+  };
+
+  edopro = stdenv.mkDerivation rec {
+    pname = "edopro";
+    version = edopro-version;
```

Comment:
The version should follow the spec: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning

This is just a random commit from upstream, the actual current release is near impossible to build with nixpkgs.

Reply:
Wait, no, again, you changed that! Damn, I spent a good week trying to get `fmt` to work, didn't realize nixpkgs packaged multiple versions of it.